src/sonic-config-engine/minigraph.py

```diff
@@ -860,6 +897,14 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None):
         if port.get('speed') == '100000':
             port['fec'] = 'rs'
 
+        # If FECDisabled, any previous decision will be override
+        if port_config_file:
+            alias = port.get('alias')
+        else:
+            alias = port_name
+        if linkmetas.get(alias, {}).get("FECDisabled") == "True":
```

Comment:
case insensitive

Reply:
by default, the fec is none. the issue of this code is that if minigraph has issue, for example setting fec to none for 10G port, then it might not work as 10g does not support fec configuration.

therefore, ideally, you need to first check if a specific link speed support fec configuration or not. but that complicates your code. since the purpose here is to override the 100g fec configuration. i suggest following logic.

if port.get('speed') == '100000' and linkmetas.get(alias, {}).get("FECDisabled") != True:
    port['fec'] = 'rs'

Reply:
To clarify, if speed == 100000 and FECDisabled == True, do you expected an explicit 'none'?

---
In reply to: [422022078](https://github.com/Azure/sonic-buildimage/pull/4556#discussion_r422022078) [](ancestors = 422022078)

Reply:
no, i do not expect. I do not see such cases here now. as i commented earlier, if there is no entry in the config db, it is default disabled.